tests/greenio_test.py

```diff
@@ -856,6 +857,10 @@ class TestGreenIoLong(tests.LimitedTestCase):
     TEST_TIMEOUT = 10  # the test here might take a while depending on the OS
 
     def test_multiple_readers(self):
+        from eventlet.hubs.asyncio import Hub
+        if isinstance(get_hub(), Hub):
```

Comment:
Can't we modify this condition to catch the `RuntimeError` raised at line 119 in `eventlet.hubs/asyncio.py` and then act accordingly rather than checking the type of the instance?

That would allow to assert that the mechanism implemented into the hub are well triggered in this kind of condition.

Reply:
Mm, maybe. Ideally we'd figure out how to do multiple readers in asyncio.

Reply:
Oh also the hub is already started at this point I think? So it wouldn't actually raise a `RuntimeError` in a place that the test can actually deal with.

This also raises the issue though that one can set the flag after the hub is running, so the `RuntimeError` check is not sufficient...

Reply:
OK came up with a better approach.